swiftnav/ephemeris.pyx

```diff
@@ -0,0 +1,258 @@
+
+  property toe:
+    def __get__(self):
+        return GpsTime(self.ephemeris.toe.wn, self.ephemeris.toe.tow)
+    def __set__(self, toe):
+        self.ephemeris.toe.wn  = toe.wn
+        self.ephemeris.toe.tow = toe.tow
+
+  property toc:
+    def __get__(self):
+        return GpsTime(self.ephemeris.toc.wn, self.ephemeris.toc.tow)
+    def __set__(self, toc):
+        self.ephemeris.toc.wn  = toc.wn
+        self.ephemeris.toc.tow = toc.tow
+
+  property valid:
+    def __get__(self):
+        return self.ephemeris.valid
+    def __set__(self, valid):
+        self.ephemeris.valid = valid
+
+  property healthy:
+    def __get__(self):
+        return self.ephemeris.healthy
+    def __set__(self, healthy):
+        self.ephemeris.healthy = healthy
+
+  property prn:
+    def __get__(self):
+        return self.ephemeris.prn
+    def __set__(self, prn):
+        self.ephemeris.prn = prn
+
+def calc_sat_state(Ephemeris eph, GpsTime time):
+  cdef np.ndarray[np.double_t, ndim=1, mode="c"] pos_ = \
+    np.array([0,0,0], dtype=np.double)
+  cdef np.ndarray[np.double_t, ndim=1, mode="c"] vel_ = \
+    np.array([0,0,0], dtype=np.double)
+
+  cdef double clock_err
+  cdef double clock_rate_err
+
+  cdef ephemeris_c.ephemeris_t eph_ = eph.ephemeris
+  cdef gpstime_c.gps_time_t t_      = time.gps_time
+
+  ephemeris_c.calc_sat_state(&eph_, t_,
+                 &pos_[0], &vel_[0],
+                 &clock_err, &clock_rate_err
+                 )
+
+  return pos_, vel_, clock_err, clock_rate_err
+
+# def measure_b_with_external_ambs(alms, GpsTime timestamp,
+#                                    numpy_measurements,
+#                                    ambs,
+#                                    reciever_ecef): #TODO eventually, want to get reciever_ecef from data
+#   n = len(alms)
+#   cdef almanac_t al[32]
+#   cdef almanac_t a_
+#   cdef np.ndarray[np.uint8_t, ndim=1, mode="c"] prns = \
+#         np.empty(n, dtype=np.uint8)
+#   for i, a in enumerate(alms):
+#     a_ = (<Almanac> a).almanac
+#     memcpy(&al[i], &a_, sizeof(almanac_t))
+#     prns[i] = (<Almanac> a).prn
+
+#   cdef np.ndarray[np.double_t, ndim=1, mode="c"] ref_ecef_ = \
+#     np.array(reciever_ecef, dtype=np.double)
+
+#   cdef np.ndarray[np.double_t, ndim=1, mode="c"] ambs_ = \
+#     np.array(ambs, dtype=np.double)
+
+#   cdef gps_time_t timestamp_ = timestamp.gps_time
+
+#   cdef sdiff_t sdiffs[32]
+#   almanacs_to_single_diffs(len(alms), &al[0], timestamp_, sdiffs)
+
+#   for i, (l,c) in enumerate(numpy_measurements):
+#     sdiffs[i].pseudorange = c
+#     sdiffs[i].carrier_phase = l
+  
+#   cdef np.ndarray[np.double_t, ndim=1, mode="c"] b = \
+#     np.empty(3, dtype=np.double)
+
+#   dgnss_management_c.measure_b_with_external_ambs(&ref_ecef_[0],
+#                                                   n, &sdiffs[0],
+#                                                   &ambs_[0],
+#                                                   &b[0])
+#   return b
```

Comment:
Hi there again, your friendly neighborhood Git reviewer just checkin in to see if you folks meant to put up all these comments here! The homeowners association board members makes SUCH a fuss when we do that after all. Anyway, have a nice day!